mmpose/datasets/pipelines/bottom_up_transform.py

```diff
@@ -226,22 +226,103 @@ def __call__(self, joints):
-        output_res = self.output_res
+        output_size = self.output_size
         for i in range(len(joints)):
             tot = 0
             for idx, pt in enumerate(joints[i]):
                 x, y = int(pt[0]), int(pt[1])
-                if (pt[2] > 0 and 0 <= y < self.output_res
-                        and 0 <= x < self.output_res):
+                if (pt[2] > 0 and 0 <= y < self.output_size
+                        and 0 <= x < self.output_size):
                     if self.tag_per_joint:
                         visible_kpts[i][tot] = \
-                            (idx * output_res**2 + y * output_res + x, 1)
+                            (idx * output_size**2 + y * output_size + x, 1)
                     else:
-                        visible_kpts[i][tot] = (y * output_res + x, 1)
+                        visible_kpts[i][tot] = (y * output_size + x, 1)
                     tot += 1
         return visible_kpts
 
 
+class PAFGenerator:
+    """Generate part affinity fields.
+
+    Args:
+        output_size (int): Size of feature map.
+        limb_width (int): Limb width of part affinity fields.
+        skeleton (list[list]): connections of joints.
+    """
+
+    def __init__(self, output_size, limb_width, skeleton):
+        self.output_size = output_size
+        self.limb_width = limb_width
+        self.skeleton = skeleton
+
+    def _accumulate_paf_map(self, pafs, src, dst, count):
+        """Accumulate part affinity fields between two given joints.
+
+        Args:
+            pafs (np.ndarray[2xHxW]): paf maps (2 dimensions:x axis and
+                y axis) for a certain limb connection.
+            src (np.ndarray[2,]): coordinates of the source joint.
+            dst (np.ndarray[2,]): coordinates of the destination joint.
+            count (np.ndarray[HxW]): count map that preserves the number of
+                non-zero vectors at each point.
+        """
+        limb_vec = dst - src
+        norm = np.linalg.norm(limb_vec)
+        if norm == 0:
+            unit_limb_vec = np.zeros(2)
+        else:
+            unit_limb_vec = limb_vec / norm
+
+        min_x = max(np.floor(min(src[0], dst[0]) - self.limb_width), 0)
+        max_x = min(
+            np.ceil(max(src[0], dst[0]) + self.limb_width),
+            self.output_size - 1)
+        min_y = max(np.floor(min(src[1], dst[1]) - self.limb_width), 0)
+        max_y = min(
+            np.ceil(max(src[1], dst[1]) + self.limb_width),
+            self.output_size + 1)
+
+        range_x = list(range(int(min_x), int(max_x + 1), 1))
+        range_y = list(range(int(min_y), int(max_y + 1), 1))
+        xx, yy = np.meshgrid(range_x, range_y)
+        delta_x = xx - src[0]
+        delta_y = yy - src[1]
+        dist = np.abs(delta_x * unit_limb_vec[1] - delta_y * unit_limb_vec[0])
+        mask_local = (dist < self.limb_width)
+        mask = np.zeros_like(count, dtype=bool)
+        mask[xx, yy] = mask_local
+
+        pafs[0, mask] += unit_limb_vec[0]
```

Comment:
Okay `pafs` is also modified inplace. From its usage at L317, I suggest the following changes.

* Add sentence `This argument will be modified inplace` to L264 and L268.
* Recover L298 back to `count += mask`
* Change function name to `_accumulate_paf_map_` where the trailing `_` means inplace
* Change L316-318 to `self._accumulate_paf_map(pafs[2 * idx:2 * idx + 2], src[:2], dst[:2], count)`